src/lib.rs

```diff
@@ -432,6 +460,44 @@ where
     }
 }
 
+#[derive(Debug)]
+/// Iterator over tokens sorted by rank given a token (sorted by highest probability)
+pub struct RankIterator<'a, T>
```

Comment:
Maybe rename to something more specific? Maybe `ProbabilityIterator` or `SortedRankIterator`?